remote-api/src/main/java/org/alfresco/rest/framework/resource/parameters/where/BasicQueryWalker.java

```diff
@@ -0,0 +1,259 @@
+    protected boolean validateStrictly = true;
+
+    public BasicQueryWalker()
+    {
+        this.properties = new HashMap<>();
+    }
+
+    public BasicQueryWalker(final String... expectedPropertyNames)
+    {
+        this();
+        this.expectedPropertyNames = Set.of(expectedPropertyNames);
+    }
+
+    public BasicQueryWalker(final Collection<String> expectedPropertyNames)
+    {
+        this();
+        this.expectedPropertyNames = expectedPropertyNames;
+    }
+
+    public void setClausesNegatable(final boolean clausesNegatable)
+    {
+        this.clausesNegatable = clausesNegatable;
+    }
+
+    public void setValidateStrictly(boolean validateStrictly)
+    {
+        this.validateStrictly = validateStrictly;
+    }
+
+    @Override
+    public void exists(String propertyName, boolean negated)
+    {
+        verifyPropertyExpectedness(propertyName);
+        verifyClausesNegatability(negated, propertyName);
+        addProperties(propertyName, WhereClauseParser.EXISTS, negated);
+    }
+
+    @Override
+    public void between(String propertyName, String firstValue, String secondValue, boolean negated)
+    {
+        verifyPropertyExpectedness(propertyName);
+        verifyClausesNegatability(negated, propertyName);
+        addProperties(propertyName, WhereClauseParser.BETWEEN, negated, firstValue, secondValue);
+    }
+
+    @Override
+    public void comparison(int type, String propertyName, String propertyValue, boolean negated)
+    {
+        verifyPropertyExpectedness(propertyName);
+        verifyClausesNegatability(negated, propertyName);
+        if (WhereClauseParser.EQUALS == type && isAndSupported() && containsProperty(propertyName, WhereClauseParser.IN, negated))
+        {
+            throw new InvalidQueryException(String.format(EQUALS_AND_IN_NOT_ALLOWED_TOGETHER, propertyName));
+        }
+
+        addProperties(propertyName, type, negated, propertyValue);
+    }
+
+    @Override
+    public void in(String propertyName, boolean negated, String... propertyValues)
+    {
+        verifyPropertyExpectedness(propertyName);
+        verifyClausesNegatability(negated, propertyName);
+        if (isAndSupported() && containsProperty(propertyName, WhereClauseParser.EQUALS, negated))
+        {
+            throw new InvalidQueryException(String.format(EQUALS_AND_IN_NOT_ALLOWED_TOGETHER, propertyName));
+        }
+
+        addProperties(propertyName, WhereClauseParser.IN, negated, propertyValues);
+    }
+
+    @Override
+    public void matches(final String propertyName, String propertyValue, boolean negated)
+    {
+        verifyPropertyExpectedness(propertyName);
+        verifyClausesNegatability(negated, propertyName);
+        addProperties(propertyName, WhereClauseParser.MATCHES, negated, propertyValue);
+    }
+
+    @Override
+    public void and()
+    {
+        // Don't need to do anything here - it's enough to enable AND operator.
+        // OR is not supported at the same time.
```

Comment:
Higher up you seemed to say the opposite - that OR is supported, but AND isn't.

Reply:
Class `BasicQueryWalker` by default supports `AND` operations, but it's overwritten by:
`QueryHelper
	.resolve(namesQuery)
	.usingOrOperator()`
To support `OR` operations instead.

Reply:
Ah - I missed that. Thanks!